src/views/mod.rs

```diff
@@ -119,6 +129,145 @@ pub mod krate_publish;
 pub use self::krate_publish::CrateDependency as EncodableCrateDependency;
 pub use self::krate_publish::NewCrate as EncodableCrateUpload;
 
+/// Information about whether this version built on the specified Rust version and target, as
+/// uploaded by the `cargo publish-build-info` command.
+#[derive(Debug, Serialize, Deserialize)]
+pub struct EncodableVersionBuildInfoUpload {
+    pub rust_version: EncodableRustChannelVersion,
+    pub target: String,
+    pub passed: bool,
+}
+
+/// Aggregated build info for a crate version grouped by Rust channel for front-end display
+/// convenience.
+#[derive(Debug, Serialize, Deserialize, Default)]
+pub struct EncodableVersionBuildInfo {
+    pub id: i32,
+    pub ordering: HashMap<String, Vec<String>>,
+    pub stable: HashMap<String, HashMap<String, bool>>,
+    pub beta: HashMap<NaiveDate, HashMap<String, bool>>,
+    pub nightly: HashMap<NaiveDate, HashMap<String, bool>>,
+}
+
+/// `MaxBuildInfo` in JSON form.
+#[derive(Debug, Default, Serialize, Deserialize)]
+pub struct EncodableMaxVersionBuildInfo {
```

Comment:
Every field on `MaxBuildVersionInfo` implements `Serialize`, doesn't it? Is there a reason we need this as a separate struct?